trade-finance-manager-ui/.eslintrc.js

```diff
@@ -59,7 +59,10 @@ module.exports = {
         project: './tsconfig.eslint.json',
         tsconfigRootDir: __dirname,
       },
-      rules: baseRules,
+      rules: {
+        ...baseRules,
+        '@typescript-eslint/no-misused-promises': 'off',
```

Comment:
Same again

Reply:
This one is actually needed as a result of changes the utilisation reports routes file to typescript. Our controllers technically do not match the definition of what express expects the controllers to be: express expects a function with the `(req, res, next?)` parameters and returning `void`, but our functions are usually `async`, meaning they return `Promise<void>` (sometimes they return other things, but this isn't really a problem as we could just move the `return` statement to the next line and everything would work the same). Typescript type-checking isn't annoyed by this, but the typescript linter is as it regards functions returning `Promise<void>` where `void` was expected as a misused promise (hence the name)

Reply:
Hmm interesting... I wonder if we should actually disable this for this specific case, whether that is doing inline `// eslint-disable-line @typescript-eslint/no-misused-promises` or if we are able to do something clever in the eslint config to apply a specific rule to route files. It just seems like a rule that is otherwise generally useful so feels a shame to turn it off otherwise?

Is it right that it is ok having async controllers returning promises because our controllers are the last thing our routes call to?

Reply:
I'm not sure it is necessarily worthwhile to try to keep the eslint rule enabled because I'm not sure that it provides anything useful. Its main utility seems to be for situations where you are wrapping everything with promises when you don't need to and, even though that code will run, there is an overhead to building code like that. Do you have any situations in mind where this rule could catch something?

Also, I believe it's fine for both middleware and the controller to return `Promise<void>` instead of `void` because everything in express seems to execute asynchronously (I will look into this now and leave a follow up comment)

Reply:
From [the docs](https://typescript-eslint.io/rules/no-misused-promises/) it seems it helps you spot places you have forgotten an await e.g. if you try and spread the result of an api call into a new object without awaiting then it will give you the no-misused-promises warning, or if you try to use a Promise<boolean> inside an if statement having forgotten to await it.

Reply:
I've had a read also and agree their docs do suggest the actual returning of the promise is fine in this case, not sure why express has typed it as such...

Reply:
I think that side of the benefits from `no-misused-promises` is also captured by the [`no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises/) rule. Can you think of any cases where `no-floating-promises` will miss something that `no-misused-promises` would catch?

Reply:
This is obviously rather manufactured but if you turn of no-misused-promises this doesn't show any error:
```
const reportsForPeriodPromise = api.getUtilisationReports(bankId, {
    reportPeriod,
  });

  const obj = {
    ...reportsForPeriodPromise
  };

  const reportsForPeriod = await reportsForPeriodPromise;
```

Reply:
Yeh that is interesting - the message you get there is `Expected a non-Promise value to be spreaded in an object` which is a strange one.

Do you think disabling this rule only for files in the `server/routes` directory would be a good compromise here?